tests/func/test_repro.py

```diff
@@ -1420,3 +1420,45 @@ def _format_dvc_line(line):
 
 def _hide_md5(text):
     return re.sub(r"\b[a-f0-9]{32}\b", "<md5>", text)
+
+
+class TestReproDownstream(TestDvc):
+    def test(self):
+        # The dependency graph should look like this:
+        #
+        #       E
+        #      / \
+        #     D   F
+        #    / \   \
+        #   B   C   G
+        #    \ /
+        #     A
+        #
+        assert main(["run", "-o", "A", "echo A>A"]) == 0
+        assert main(["run", "-d", "A", "-o", "B", "echo B>B"]) == 0
+        assert main(["run", "-d", "A", "-o", "C", "echo C>C"]) == 0
+        assert main(["run", "-d", "B", "-d", "C", "-o", "D", "echo D>D"]) == 0
+        assert main(["run", "-o", "G", "echo G>G"]) == 0
+        assert main(["run", "-d", "G", "-o", "F", "echo F>F"]) == 0
+        assert main(["run", "-d", "D", "-d", "F", "-o", "E", "echo E>E"]) == 0
+
+        # We want the evaluation to move from B to E
+        #
+        #       E
+        #      /
+        #     D
+        #    /
+        #   B
+        #
+        evaluation = self.dvc.reproduce("B.dvc", downstream=True, force=True)
+
+        assert len(evaluation) == 3
+        assert evaluation[0].relpath == "B.dvc"
+        assert evaluation[1].relpath == "D.dvc"
+        assert evaluation[2].relpath == "E.dvc"
+
+    def test_cli(self):
```

Comment:
This is not the way to test CLI I was talking about. Something like:

```
args.downstream = True
cmd = CmdRepro(args)
with mock.patch.object(cmd.repo, "reproduce") as mock_repo:
     # check that repro was called with `downstream=True`
args.downstream = False
with mock.patch.object(cmd.repo, "reproduce") as mock_repo:
     # check that repro was called with `downstream=False`
```
and it should be in `tests/unit/command/test_repro.py`.

Reply:
got it, @efiop :+1: